### PR TITLE
Add more general version of dmap_uni

### PR DESCRIPTION
**Pull request by asifmallik**
The dmap_uni lemma requires that the function is injective instead of simply being injective with respect to the support of the distribution so I proved a lemma where that is the assumption.

### CODE REVIEW

File: theories/distributions/Distr.ec [hunk 1]
```diff
@@ -784,15 +784,21 @@ lemma dmap_ll (d : 'a distr) (f : 'a -> 'b) :
   is_lossless d => is_lossless (dmap d f).
 proof. by rewrite /is_lossless weight_dmap. qed.
 
-lemma dmap_uni (d : 'a distr) (f : 'a -> 'b) : 
-  injective f => is_uniform d => is_uniform (dmap d f).
-proof. 
+lemma dmap_unig (d : 'a distr) (f : 'a -> 'b) :
+  (forall (x, y : 'a),  0%r < mu1 d y => f x = f y => x = y) => is_uniform d => is_uniform (dmap d f).
```
Comment (strub) [inline theories/distributions/Distr.ec]: Use the `_ \in _` predicate.

File: theories/distributions/Distr.ec [hunk 2]
```diff
@@ -784,15 +784,21 @@ lemma dmap_ll (d : 'a distr) (f : 'a -> 'b) :
   is_lossless d => is_lossless (dmap d f).
 proof. by rewrite /is_lossless weight_dmap. qed.
 
-lemma dmap_uni (d : 'a distr) (f : 'a -> 'b) : 
-  injective f => is_uniform d => is_uniform (dmap d f).
-proof. 
+lemma dmap_unig (d : 'a distr) (f : 'a -> 'b) :
```
Comment (strub) [inline theories/distributions/Distr.ec]: `dmap_unig` -> `dmap_uni_in_inj`

Comment (asifmallik) [reply theories/distributions/Distr.ec]: @strub How about `dmap_uni_support` instead in the spirit of `mu_eq_support`?

**Comment by asifmallik** [comment]
I have generalized the lemma further by adding another assumption. However, that required me to change the proof completely.

**Pull request merged**